src/labone/nodetree/node.py

```diff
@@ -0,0 +1,863 @@
+        self._value_structure: TreeProp = value_structure
+
+    def try_generate_subnode(self, next_path_segment: NormalizedPathSegment):
+        extended_path = (*self._path_segments, next_path_segment)
+
+        try:
+            next_subtree = self._subtree_structure[next_path_segment]
+        except KeyError:
+            if next_path_segment == WILDCARD:
+                raise RuntimeError(f'Wildcards * All are not allowed in a tree '
+                                   f'representing measurement results')
+
+            raise RuntimeError(
+                f"node path_segments '{self}' cannot be extended with /{next_path_segment}\n"
+                f"viable extentions would be {list(self._subtree_structure.keys())}")
+        else:
+            deeper_node = ResultNode(extended_path,
+                              self._subtree_structure[next_path_segment],
+                              self._value_structure,
+                              self._path_aliases)
+
+            if next_subtree == {}:
+                # give value instead of subnode if already at a leaf
+                # generate hypothetical node in order to apply normal behaviour,
+                # including path alias redirection
+                return self._value_structure.get(deeper_node) #.get_with_segments(extended_path)
+
+            return deeper_node
+
+    def __dir__(self) -> t.Iterable[str]:
+        return [pythonify_path_segment(p) for p in self._subtree_structure.keys()]
+
+    def __call__(self, *args, **kwargs):
+        raise RuntimeError('Trying to get/set on a result node, which is not possible. This node only represents '
+                           'values of a former get-request. To interact with a device, '
+                           'make sure to operate on normal nodes.')
+
+
+class Node(MetaNode):
+    _matches_index = UniqueCounter()
+
+    def __init__(self,
+                 tree_manager: NodeTreeManager,
+                 path_segments: t.Tuple[t.Union[str, int]],
+                 subtree_structure: NestedDict | None = None,
+                 path_aliases: dict[tuple[NormalizedPathSegment], tuple[NormalizedPathSegment]] | None = None
+                 ):
+
+        if path_aliases is None:
+            path_aliases = {}
+        if subtree_structure is None:
+            try:
+                subtree_structure = nested_dict_access(path_segments, tree_manager.structure)
+            except KeyError:
+                KeyError(f"cannot find {path_segments} in tree-structure. Make "
+                         f"sure the path_segments is valid or pass a custom "
+                         f"subtree_structure")
+
+        super().__init__(path_segments, subtree_structure, path_aliases)
+        self._tree_manager = tree_manager
+        self._queue: DataQueue | None = None
+
+    def __eq__(self, other: 'Node'):
+        return (self._path_segments == other._path_segments
```

Comment:
You must also check the underlying session 🙈 ... the user can create two sessions for the same device and have two separate node trees for each session.